src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPProgressListener.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using System.Composition;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
+{
+    [Shared]
+    [Export(typeof(LSPProgressListener))]
+    internal class DefaultLSPProgressListener : LSPProgressListener, IDisposable
```

Comment:
This implementation of `LSPProgressListener` works well for our current FAR purposes. It'll likely need to be adapted in the future to be able to listen to progress notifications which don't originate from a handler which needs to be blocked till the notification is processed (diagnostics come to mind). 

While we could update this implementation to account for such notification event types, the changes should remain minimal, and I would rather not pre-optimize for a requirement which may not be totally clear / flushed out yet.

Just putting this note up as something to keep in mind.